### PR TITLE
cli now can print details

### PR DESCRIPTION
**Pull request by judovana**

### CODE REVIEW

**Comment by AurumTheEnd** [comment]
@judovana I was just reviewing it... nvm. You merge too fast all the time :/. Not compatible with the way I sporadically work during school.

**Review by AurumTheEnd — COMMENTED** [review]
:man_shrugging:

File: runtime-decompiler/src/main/java/org/jrd/backend/data/Cli.java [hunk 1]
```diff
@@ -659,16 +673,25 @@ private void listClasses() throws IOException {
...
+            VmManager vmManager,
+            List<Pattern> filter,
+            boolean details) throws IOException {
+        List<ClassInfo> allClasses;
+        if (details) {
+            allClasses = Arrays.stream(obtainClassesDetails(vmInfo, vmManager)).collect(Collectors.toList());
+        } else {
+            allClasses = Arrays.stream(obtainClasses(vmInfo, vmManager))
+                    .map(a -> new ClassInfo(a, null, null))
+                    .collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (judovana) [reply runtime-decompiler/src/main/java/org/jrd/backend/data/Cli.java:690]: I agree it is nasty. But the method isquite used around, so all the consuming classes would eet to map. That swhy I put it here.

File: runtime-decompiler/src/main/java/org/jrd/backend/data/Cli.java [hunk 2]
```diff
@@ -777,18 +799,27 @@ private static String invalidityToString(boolean invalidWrapper) {
     }
 
     public static String[] obtainClasses(VmInfo vmInfo, VmManager manager) {
-        AgentRequestAction request = DecompilationController.createRequest(
-                vmInfo, AgentRequestAction.RequestAction.CLASSES, null
-        );
+        AgentRequestAction.RequestAction requestType = AgentRequestAction.RequestAction.CLASSES;
+        AgentRequestAction request = DecompilationController.createRequest(vmInfo, requestType, null);
```
[the remark this answers is not in the document]
Comment (judovana) [reply runtime-decompiler/src/main/java/org/jrd/backend/data/Cli.java:803]: Yah. I originallyhad the main swithc on details here. Butthen abandoned it and left this as it is .

**Pull request merged**